zeppelin-web/src/components/data-validator/data-validator-factory.js

```diff
@@ -0,0 +1,97 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+angular.module('zeppelinWebApp').factory('DataValidator', function($rootScope) {
+
+  var msg = '';
+  var errorStatus = false;
+  var DataValidator = function(schema) {
+    this.schema = schema;
+    this.error = getErrorStatus;
+    this.setError = setError;
+    this.checkData = checkData;
+    this.getMsg = getMsg;
+    this.setMsg = setMsg;
+    this.data = null;
+  };
+
+  function checkData() {
+    if(basicCheck(this.data, this.schema)) { // jshint ignore:line
+      msg += 'data is exisiting | ';
+    } else {
+      msg += 'data does not exisiting | ';
+    }
+  }
+
+  function getMsg() {
+    return msg;
+  }
+
+  function setMsg(newMsg) {
+    msg += newMsg;
+  }
+
+  function getErrorStatus() {
+    return errorStatus;
+  }
+
+  function setError() {
+    errorStatus = true;
+  }
+
+  function basicCheck(data, schema) {
+    if(data.code && data.rows) {
+      rowCheck(data.rows, 3, schema);
```

Comment:
@felixcheung : 
With updated code `basicCheck`[1] function return bool and and it is used in `checkData()`[2].

`rowCheck()` do not return any bool at now. When ever you found validator issue you can call new function `setError();`. It will handle it by making `errorStatus` to true. I will added this function for documentation too. So I will be easy for new developers who will writting new data valitors. 

[1] https://github.com/Madhuka/incubator-zeppelin/blob/milestone05-map/zeppelin-web/src/components/data-validator/data-validator-factory.js#L59
[2] https://github.com/Madhuka/incubator-zeppelin/blob/milestone05-map/zeppelin-web/src/components/data-validator/data-validator-factory.js#L32